File: LandmarkBookjava/app/src/main/java/com/gorkemozcan/landmarkbookjava/DetailsActivity.java
package com.gorkemozcan.landmarkbookjava;



import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.gorkemozcan.landmarkbookjava.databinding.ActivityDetailsBinding;

import org.w3c.dom.Text;

public class DetailsActivity extends AppCompatActivity {
    private ActivityDetailsBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityDetailsBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);
        //inflate -> xml ile kodumuzu bağlamaya çalıştığımmız yer
        Intent intent = getIntent();

        Singleton singleton = Singleton.getInstance();
        Landmark selectedLandmark = singleton.getSentLandmark();


        //casting bişeyi bişeye dönüştürmek
        //Landmark selectedLandmark = (Landmark) intent.getSerializableExtra("Landmark");
        binding.nameText.setText(selectedLandmark.name);
        binding.countryText.setText(selectedLandmark.country);
        binding.imageView.setImageResource(selectedLandmark.image);
    }
}

File: LandmarkBookKotlin/app/src/main/java/com/gorkemozcan/landmarkbookkotlin/DetailsActivity.kt
package com.gorkemozcan.landmarkbookkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.gorkemozcan.landmarkbookkotlin.databinding.ActivityDetailsBinding

class DetailsActivity : AppCompatActivity() {

    private lateinit var binding:ActivityDetailsBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailsBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        val intent = intent
        //Casting

       // val selectedLandmark = intent.getSerializableExtra("landmark") as Landmark

        //val selectedLandmark = chosenLandmark
        /*
        binding.nameText.text = selectedLandmark!!.name
        binding.countryText.text = selectedLandmark!!.country
        binding.imageView.setImageResource(selectedLandmark!!.image)

         */
        val selectedLandmark = mySingleton.chosenLandmark
        selectedLandmark?.let {
            binding.nameText.text = it.name
            binding.countryText.text = it.country
            binding.imageView.setImageResource(it.image)
        }


    }
}